src/main/java/com/researchspace/dao/customliquibaseupdates/CryptingApiKeysAndUpdateDatabase_RSDEV310.java

```diff
@@ -0,0 +1,40 @@
+package com.researchspace.dao.customliquibaseupdates;
+
+import com.researchspace.core.util.CryptoUtils;
+import com.researchspace.dao.UserApiKeyDao;
+import com.researchspace.model.UserApiKey;
+import java.util.List;
+import liquibase.database.Database;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class CryptingApiKeysAndUpdateDatabase_RSDEV310 extends AbstractCustomLiquibaseUpdater {
+
+  private UserApiKeyDao userApiKeyDao;
+  private int recordsUpdated = 0;
+
+  @Override
+  protected void addBeans() {
+    userApiKeyDao = context.getBean(UserApiKeyDao.class);
+  }
+
+  @Override
+  public String getConfirmationMessage() {
+    return "Updated " + recordsUpdated + " records";
+  }
+
+  @Override
+  protected void doExecute(Database database) {
+
+    List<UserApiKey> userApiKeyListToCheck = userApiKeyDao.getAll();
+
+    userApiKeyListToCheck.stream()
+        .filter(userKey -> userKey.getApiKey().length() <= 32)
```

Comment:
becasue we want to crypt only the old clear apiKeys that were 32 char long.
Basically it is just a check on what we are crypting to avoid double-hash keys that are already hashed